### DISCUSSION TITLE
Password/Confirm password fields don't work, cannot type after validation fails

### DISCUSSION THREAD

**Discussion by mmvcode** [react-hook-form/react-hook-form#2425] [Q&A] [answered]
I have the following form:
[https://codesandbox.io/s/adoring-moon-o4onq?file=/src/App.js](https://codesandbox.io/s/adoring-moon-o4onq?file=/src/App.js)

The form is a basic reset password form with Passowrd/Confirm password form with a custom [TextInput](https://codesandbox.io/s/adoring-moon-o4onq?file=/src/TextInput.js).

**Issue**: When clicking submit, the validation will fail with "Field required" error, even if the inputs was 8 or more characters. Moreover, after submitting and validation fails, I cannot type in the fields anymore.

This is likely a "ref" issue and I believe the error is somewhere in https://codesandbox.io/s/adoring-moon-o4onq?file=/src/TextInput.js at line ~241 (IconRight), coming from 276 (WithIcon). I am kind of new to refs and I am not quite sure how to fix it. Any ideas?

Thanks.

**Comment by bluebill1049** [comment, marked as answer]
it's not a ref issue, it's an issue within your https://codesandbox.io/s/adoring-moon-o4onq?file=/src/TextInput.js put a console.log in each those little sub-component, you are re-mounting them each re-render. you should put those subcomponent into its own component.

**Reply by mmvcode** [reply]
Will try, thanks 

**EDIT**: Yes, that was the issue. Thanks again.
